server/pulp/server/managers/consumer/applicability.py

```diff
@@ -118,29 +121,48 @@ def regenerate_applicability_for_repos(repo_criteria):
         repo_criteria.fields = ['id']
         repo_ids = [r.repo_id for r in model.Repository.objects.find_by_criteria(repo_criteria)]
 
+        task_group_id = uuid4()
+
         for repo_id in repo_ids:
-            # Find all existing applicabilities for given repo_id. Setting batch size of 5 ensures
-            # the MongoDB cursor does not time out. See https://pulp.plan.io/issues/998#note-6 for
-            # more details.
-            existing_applicabilities = RepoProfileApplicability.get_collection().find(
-                {'repo_id': repo_id}).batch_size(5)
-            for existing_applicability in existing_applicabilities:
+            existing_applicability_ids = RepoProfileApplicability.get_collection().find(
+                {'repo_id': repo_id}, {'_id': 1})
+            for batch in paginate(existing_applicability_ids, 100):
+                batch_regenerate_applicability_task.apply_async((repo_id, batch),
+                                                                **{'group_id': task_group_id})
+        return task_group_id
+
+    @staticmethod
+    def batch_regenerate_applicability(repo_id, existing_applicability_ids, group_id=None):
+        """
+        Regenerate and save applicability data for a batch of existing applicabilities
+
+        :param repo_id: Repository id for which applicability is being calculated
+        :type repo_id: str
+        :param existing_applicability_ids: Tuple of Object Ids for applicability profiles
+        :type existing_applicability_ids: tuple of dicts in form of {"_id": ObjectID('mongo-id')}
+        :param group_id: task group id that this task should belong to
+        :type group_id: UUID
```

Comment:
We are using a UUID field type in the TaskStatus model. So we are expecting this to be of type UUID 4.